src/app/config.rs

```diff
@@ -36,16 +55,19 @@ pub struct FeesConfig {
     pub mempool_space: MempoolSpaceConfig,
 }
 
-#[derive(Debug, Clone, Serialize, Deserialize, Default)]
-pub struct AppConfig {
-    #[serde(default)]
-    pub blockchain: BlockchainConfig,
-    #[serde(default)]
-    pub jobs: JobsConfig,
-    #[serde(default)]
-    pub signer_encryption: SignerEncryptionConfig,
-    #[serde(default)]
-    pub fees: FeesConfig,
+#[derive(Debug, Clone, Default, Serialize, Deserialize)]
+pub struct SecurityConfig {
+    blocked_addresses: Vec<bitcoin::Address>,
```

Comment:
I know I said we should use a Vec for deserialisation but actually we really need a hash set for efficient lookup. You'll have to figure out how that can be handled during deserialisation.